### PR TITLE
Merge 3.1.x into 3.2.x

### PR DESCRIPTION
**Pull request by morozov**

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

**Review by greg0ire — APPROVED** [review]

**Comment by morozov** [comment]
@derrabus we normally don't merge the "merge up" pull requests since there's already a merge commit. GitHub allows pushing the approved and built merge commit upstream even if the target branch is protected.

A benefit of doing that is that one can create multiple "merge up" pull requests at a time (e.g. `2.13.x` → `3.1.x` → `3.2.x` → `4.0.x`) and get them approved and merged all at once.

**Comment by derrabus** [comment]
I see, thanks for explaining.

**Pull request merged**
